tests/unit/lib/lore_test.sh

```diff
@@ -889,6 +901,104 @@ function test_process_representative_patches_duplicated_patches()
+
+  # Case 2: In reply is patch from same patchset, but not cover letter
+  # shellcheck disable=SC2317
+  function get_raw_lore_message()
+  {
+    printf 'Subject: [PATCH v2 2/3] Some title'
+  }
+  representative_patches=()
+  processed_representative_patches=()
+  REPRESENTATIVE_PATCHES_PROCESSED=0
+  process_representative_patches 'individual_patches'
+
+  assert_equals_helper 'Wrong size of `representative_patches` array' "$LINENO" 1 "${#representative_patches[@]}"
+  assert_equals_helper 'Wrong size of `processed_representative_patches` hashtable' "$LINENO" 1 "${#processed_representative_patches[@]}"
+  assert_equals_helper 'Wrong value of `REPRESENTATIVE_PATCHES_PROCESSED`' "$LINENO" 1 "$REPRESENTATIVE_PATCHES_PROCESSED"
+  assert_equals_helper 'Wrong representative patch 0' "$LINENO" "$patch1" "${representative_patches[0]}"
+  assert_equals_helper 'Representative patch 0 should be marked in hashtable' "$LINENO" 1 "${processed_representative_patches['id1']}"
+
+  # Case 3: In reply is patch from same patchset and cover letter
+  # shellcheck disable=SC2317
+  function get_raw_lore_message()
+  {
+    printf 'Subject: [PATCH v2 0/3] Some title'
+  }
+  representative_patches=()
+  processed_representative_patches=()
+  REPRESENTATIVE_PATCHES_PROCESSED=0
+  process_representative_patches 'individual_patches'
+
+  assert_equals_helper 'Wrong size of `representative_patches` array' "$LINENO" 0 "${#representative_patches[@]}"
+  assert_equals_helper 'Wrong size of `processed_representative_patches` hashtable' "$LINENO" 0 "${#processed_representative_patches[@]}"
+  assert_equals_helper 'Wrong value of `REPRESENTATIVE_PATCHES_PROCESSED`' "$LINENO" 0 "$REPRESENTATIVE_PATCHES_PROCESSED"
+
+  # Case 4: In reply isn't patch from same patchset and isn't cover letter
+  # shellcheck disable=SC2317
+  function get_raw_lore_message()
+  {
+    printf 'Subject: [PATCH 0/3] Some title'
+  }
+  representative_patches=()
+  processed_representative_patches=()
+  REPRESENTATIVE_PATCHES_PROCESSED=0
+  process_representative_patches 'individual_patches'
+
+  assert_equals_helper 'Wrong size of `representative_patches` array' "$LINENO" 1 "${#representative_patches[@]}"
+  assert_equals_helper 'Wrong size of `processed_representative_patches` hashtable' "$LINENO" 1 "${#processed_representative_patches[@]}"
+  assert_equals_helper 'Wrong value of `REPRESENTATIVE_PATCHES_PROCESSED`' "$LINENO" 1 "$REPRESENTATIVE_PATCHES_PROCESSED"
+  assert_equals_helper 'Wrong representative patch 0' "$LINENO" "$patch1" "${representative_patches[0]}"
+  assert_equals_helper 'Representative patch 0 should be marked in hashtable' "$LINENO" 1 "${processed_representative_patches['id1']}"
+
+  # Case 5: In reply isn't patch from same patchset, but is cover letter
+  # shellcheck disable=SC2317
+  function get_raw_lore_message()
+  {
+    printf 'Subject: [PATCH 0/3] Some title'
```

Comment:
I believe it is worth to test:
1. Very large subjects.
2. Very small subjects.
3. Multi-line subject (if it makes sense)
4. Subjects that have Æ.

Reply:
After thinking further,  I don't think we need to test those cases in this context, but we surely must handle subjects that have Æ in processing patches (which is a more critical context). I will create an issue for that.